src/pubsub.ts

```diff
@@ -897,7 +906,12 @@ export class PubSub {
     if (!hasProjectId && !this.isEmulator) {
       this.auth.getProjectId((err, projectId) => {
         if (err) {
-          callback(err);
+          const serviceError = Object.assign(err, {
```

Comment:
Just a nit, but maybe we should create a custom `UnknownError` for instances like this?